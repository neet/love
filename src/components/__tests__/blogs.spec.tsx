import * as React from 'react';
import * as ReactTestRenderer from 'react-test-renderer';
import Blogs from '../blogs';

const posts = [
  {
    id: '94d9b4a2f28c',
    title: 'あ',
    createdAt: '2018-05-25',
    virtuals: {
      subtitle: '',
      previewImage: {
        imageId: '',
      },
    },
    uniqueSlug: 'あ-94d9b4a2f28c',
  },
];

describe('Blogs', () => {
  it('renders a medium posts with given props', () => {
    const component = ReactTestRenderer.create(<Blogs posts={posts} />);
    const tree      = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
