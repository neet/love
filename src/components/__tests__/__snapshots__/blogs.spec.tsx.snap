// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blogs renders a medium posts with given props 1`] = `
<ul
  className="blogs section"
  role="listbox"
>
  <li
    aria-posinset={1}
    aria-setsize={1}
    className="blog"
    role="listitem"
  >
    <a
      className="blog__link"
      href="https://medium.com/@neetshin/あ-94d9b4a2f28c"
    >
      <div
        className="blog__thumbnail blog__thumbnail--no-image"
      />
      <div
        className="blog__meta"
      >
        <h3
          className="blog__title not-special-font"
        >
          あ
        </h3>
        <time
          className="blog__date"
        >
          <i
            aria-hidden={true}
            className="far fa-clock blog__date-icon"
          />
          2018年4月25日
        </time>
      </div>
    </a>
  </li>
</ul>
`;
